### ISSUE TITLE
Changing severity of the sniff passed through `sniffs` parameter

### ISSUE THREAD

**Issue by ernilambar** [10up/plugin-check#739]
In ruleset, we do like this to change the default severity:

```
<rule ref="Generic.PHP.BacktickOperator">
  <severity>7</severity>
</rule>

```

I am not being able to change severity of the sniff when we pass `sniffs` like below.

Check: `I18n_Usage_Check`

```php
return array(
	'extensions'  => 'php',
	'standard'    => 'WordPress',
	'sniffs'      => 'WordPress.WP.I18n',
	'runtime-set' => array(
		'text_domain' => $result->plugin()->slug(),
	),
);

```

**Comment by ernilambar** [comment]
CC @swissspidy @felixarntz

**Comment by swissspidy** [comment]
> I am not being able to change severity of the sniff when we pass `sniffs` like below.

What do you mean, can't you just pass `severity` as another arg? Is something not working?

**Comment by ernilambar** [comment]
> > I am not being able to change severity of the sniff when we pass `sniffs` like below.
> 
> What do you mean, can't you just pass `severity` as another arg? Is something not working?

Passing `severity` there only filters the result and shows errors greater than that severity. It does not change the severity of the sniff `WordPress.WP.I18n`.

**Comment by swissspidy** [comment]
OK, so it seems that PHPCS allows setting the severity in the config but does not allow overriding at runtime then.

https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#changing-the-default-severity-levels
https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#setting-configuration-options

What's your use case? Why do you want to override the severity in the first place?

Maybe open a discussion topic over at the PHPCS repo for that.

**Comment by ernilambar** [comment]
Currently localhost check has severity of 8. I am developing sniff for the same check. I want to replace the current check with sniff. I am not being able to set severity 8 for the sniff. Also discussion is going on in Plugin Review team regarding increasing severity of few similar sniffs.

**Comment by swissspidy** [comment]
Aha!

Have you tried overriding the `add_result_message_for_file()` method in the check class? Then you can modify the severity.

Alternatively you could create an XML config file à la `plugin-review.xml` like `Plugin_Review_PHPCS_Check` does, but that is a bit overkill for just 1 sniff.

**Comment by ernilambar** [comment]
> Aha!
> 
> Have you tried overriding the `add_result_message_for_file()` method in the check class? Then you can modify the severity.
> 
I will try this approach.

> Alternatively you could create an XML config file à la `plugin-review.xml` like `Plugin_Review_PHPCS_Check` does, but that is a bit overkill for just 1 sniff.

Yes, I also don't like to add xml file for just one sniff.

**Comment by ernilambar** [comment]
Overriding `add_result_message_for_file()` seems to be good for achieving this. We can live with that now rather than exploring options in the upstream or creating separate ruleset file.
